presto-thrift-connector-api/src/main/java/com/facebook/presto/connector/thrift/api/datatypes/SliceType.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.thrift.api.datatypes;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.VariableWidthBlock;
+import com.facebook.presto.spi.type.Type;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import javax.annotation.Nullable;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+
+final class SliceType
```

Comment:
I'd prefer to keep it stateful. The problem I see with utility methods is that they depend on each other (like toBlock depends on numberOfRecords) and on the state of class. An object can enforce consistency in constructor which won't be possible with utility methods.
As an alternative, I can make it a base class and inherit from it. Then I can rename it to "Abstract...". Let me know if you prefer that approach to the current one with delegation.